cmd/internal/flowmode/cmd_run.go

```diff
@@ -157,6 +158,17 @@ func (fr *flowRun) Run(configFile string) error {
 		return fmt.Errorf("building tracer: %w", err)
 	}
 
+	// Enable the "telemetry.useOtelForInternalMetrics" Collector feature gate.
+	// Currently, Collector components uses OpenCensus metrics by default.
+	// Those metrics cannot be integrated with Agent Flow,
+	// so we need to always use OpenTelemetry metrics.
+	err = util.EnableOtelFeatureGates(
+		"telemetry.useOtelForInternalMetrics",
+	)
+	if err != nil {
+		return err
+	}
```

Comment:
nit: I don't particularly love having component-specific logic in cmd/internal/flowmode/cmd_run.go, which is otherwise agnostic to all the different component types. It feels harder to discover where things are happening when it's spread out beyond its package.

Reply:
I changed it back to an `init()` function for flow. I'm personally not a big fan of these init functions, but I guess it's ok here :)

Reply:
I think my solution with "init()" won't work, because the flow "init()" will also get called during static mode (because the "agent" binary also imports the otelcol files). Let me know what you think about it. If I can't think of other ideas I'll probably switch it to my non-init version.

Reply:
Services might be a good spot? Something like an Otel service?

Reply:
I agree that a service seems like a better place for it 👍

Reply:
Thanks for the feedback! I updated the PR to use a service, PTAL.